### ISSUE TITLE
DNS issue

### ISSUE THREAD

**Issue by dellablue** [hassio-addons/addon-pi-hole#4]
With the following Options:

```json
{
  "log_level": "info",
  "password": "",
  "update_lists_on_start": true,
  "web_port": 80,
  "dns_port": 53,
  "ssl": false,
  "certfile": "fullchain.pem",
  "keyfile": "privkey.pem",
  "interface": "",
  "ipv6": true,
  "ipv4_address": "",
  "ipv6_address": "",
  "virtual_host": ""
}
```

It failed to load from what appears to be a DNS issue.  When I change the port from 53 to 54 it works fine but I don't know what implications there are from changing the DNS port.  

Log:

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] 01-resolver-resolv: applying... 
[fix-attrs.d] 01-resolver-resolv: exited 0.
[fix-attrs.d] 01-sudo: applying... 
[fix-attrs.d] 01-sudo: exited 0.
[fix-attrs.d] 02-nginx: applying... 
[fix-attrs.d] 02-nginx: exited 0.
[fix-attrs.d] 03-pihole: applying... 
[fix-attrs.d] 03-pihole: exited 0.
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Hass.io Add-on: Pi-hole v0.1.1
 Network-wide ad blocking using your Hass.io instance
 From: Community Hass.io Add-ons
 By: Franck Nijhof <frenck@addons.community>
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to INFO
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] 02-updates.sh: executing... 
INFO: You are running the latest version of this add-on
[cont-init.d] 02-updates.sh: exited 0.
[cont-init.d] 10-patches.sh: executing... 
[cont-init.d] 10-patches.sh: exited 0.
[cont-init.d] 11-pihole.sh: executing... 
[cont-init.d] 11-pihole.sh: exited 0.
[cont-init.d] 12-dnsmasq.sh: executing... 
[cont-init.d] 12-dnsmasq.sh: exited 0.
[cont-init.d] 13-nginx.sh: executing... 
[cont-init.d] 13-nginx.sh: exited 0.
[cont-init.d] 14-password.sh: executing... 
WARNING: No password set! This is not recommended!
Password Removed
[cont-init.d] 14-password.sh: exited 0.
[cont-init.d] 15-hostname.sh: executing... 
[cont-init.d] 15-hostname.sh: exited 0.
[cont-init.d] 16-logfiles.sh: executing... 
[cont-init.d] 16-logfiles.sh: exited 0.
[cont-init.d] 50-gravity.sh: executing... 
dnsmasq: failed to create listening socket for port 53: Address in use
[cont-init.d] 50-gravity.sh: exited 2.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] syncing disks.
[s6-finish] sending all processes the TERM signal.
```

**Comment by mpgoebel** [comment]
Do you have DNSMASQ installed using port 53?  I had that, uninstalled DNSMASQ and everything worked in the logs, although I am still trying to get the Web UI to work and the iframe does not work either.

**Comment by frenck** [comment]
What @mpgoebel already said @dellablue... you already have something DNS-ish running somewhere, which is causing collisions.

**Comment by vehsen** [comment]
Hey, i have the **same** issue with pi-hole addon default config @ hassio.
I already disabled  all other plugins like Configurator, Duck DNS, Mosquitto, SSH server, Samba share.
But there is nothing more i installed.

> dnsmasq: failed to create listening socket for port 53: Address in use
> dnsmasq: failed to create listening socket for port 53: Address in use
> dnsmasq: failed to create listening socket for port 53: Address in use

**Comment by frenck** [comment]
@vehsen It is generally a bad idea to post your issue in a closed issue. I would strongly recommend to generally open a new issue (and refer to an older one if you'd like).

Nevertheless, I've re-opened this one for now. 

The "Address in use" error is not caused by the add-on. It is the add-on (dnsmasq to be more specific) reporting that it was unable to claim the DNS port since it is already in use by an external/ another piece of software.

I'm sorry to hear you are having problems, nevertheless, this could be a million things going on here, totally outside of the scope of this addon.

Could you describe your setup a little more? Like how did you install Hass.io? What type of device are you running it on? Software versions? (of Home Assistant, Supervisor, OS e.d.).

**Comment by vehsen** [comment]
Hey okay,
I installed _resinos-hassio-1.2-raspberrypi3.img_ 
home assistant 0.65.5
supervisor 0.98

Now I installed the hassio addon Dnsmasq.
After starting/stopping Dnsmasq the pi-hole starts flawlessly but doesn't show stats @ web-UI.
So I reinstalled pi-hole.... everything seems to work now...

Strange behavoir.

**Comment by frenck** [comment]
Hassio 1.2 has a known issue with this, this was fixed in 1.3.
